CI test test-p4c-mac / test-mac-os fails intermittently installing dependencies

I've recently started seeing a lot of CI failures in this test with:

~~~
Warning: python@3.12 3.12.6 is already installed.
The currently linked version is: 3.12.5
Warning: python@3.12 3.12.6 is already installed.
The currently linked version is: 3.12.5

Error: Cannot link python@3.12
Another version is already linked: /usr/local/Cellar/python@3.12/3.12.5
Error: Cannot link python@3.12
Another version is already linked: /usr/local/Cellar/python@3.12/3.12.5

Linking /usr/local/Cellar/python@3.12/3.12.6... 
Error: Process completed with exit code 1.
~~~

Not sure how to debug this kind of issue

--- a/.github/workflows/ci-test-mac.yml
+++ b/.github/workflows/ci-test-mac.yml
@@ -93,12 +93,6 @@ jobs:
 
       - name: Install dependencies (MacOS)
         run: |
-          # FIXME: We should not need to do this, but there are currently some complications with Github CI.
-          # See https://github.com/actions/runner-images/issues/4020.
-          # See https://github.com/actions/runner-images/issues/8838.
-          brew install python@3.12 || true
-          brew link --overwrite python@3.12
-          brew pin python@3.12
           tools/install_mac_deps.sh
 
       - name: Build (MacOS)

